src/compiler/checker.ts

```diff
@@ -18856,24 +18857,30 @@ namespace ts {
 
         function checkDeclarationInitializer(declaration: VariableLikeDeclaration) {
             const type = getTypeOfExpression(declaration.initializer, /*cache*/ true);
+            return shouldUseLiteralType(declaration) ? type : getWidenedLiteralType(type);
+        }
+
+        function shouldUseLiteralType(declaration: VariableLikeDeclaration) {

```

Comment:
`shouldKeepLiteralType` may be a better name since it only matters if you already have a literal type.